cache/models/subject.js

```diff
@@ -99,11 +99,33 @@ function attachSamples(res) {
   return redisClient.smembersAsync(subjectKey).then((aspectNames) => {
     const cmds = [];
     aspectNames.forEach((aspect) => {
-      const sampleKey = sampleStore.toKey(constants.objectType.sample,
-               res.absolutePath + '|' + aspect);
-      const aspectKey = sampleStore.toKey(constants.objectType.aspect, aspect);
-      cmds.push(['hgetall', sampleKey]);
-      cmds.push(['hgetall', aspectKey]);
+
+      if (featureToggles.isFeatureEnabled('attachSamplesPrefilterAspects')) {
+        /*
+         * If there is an aspect name filter, do it now instead of doing extra
+         * hgetall calls for aspects and samples we don't need!
+         */
+        let aspectPassesFilter = true; // default true if no aspect name filter
+        if (filters.aspect.includes && filters.aspect.includes.size) {
```

Comment:
wouldn't underscore lib help with undefined chain (_.isUndefined)? or something like that... I remember somewhere we were using a lib for that.

Reply:
There's just one usage of `lodash` outside of views (in bots db model, using "extend"), but we *do* use `just-safe-get` in some places instead of having to check for existence of many nested layers of attributes... here it wasn't so convoluted that I felt it needed a semantically cleaner way of expressing it